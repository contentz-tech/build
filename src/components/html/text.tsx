import { jsx } from "@emotion/core";
import parse from "parse-url";

function P(props: any) {
  return (
    <p
      {...props}
      css={{
        margin: "1em 0",
        fontSize: "1em"
      }}
    />
  );
}

function Blockquote(props: any) {
  return (
    <blockquote
      {...props}
      css={{
        borderLeft: "3px solid black",
        boxSizing: "border-box",
        paddingLeft: "calc(2em - 3px)",
        marginLeft: "-2em",
        marginRight: 0,
        fontWeight: "normal",
        fontSize: "1em",
        P: {
          margin: 0
        }
      }}
    />
  );
}

function Anchor(props: any) {
  const extraProps = {
    target: !parse(props.href).resource ? "_self" : "_blank",
    ...(!parse(props.href).resource ? {} : { rel: "nofollow noopener" })
  };

  return (
    <a
      {...props}
      {...extraProps}
      css={{
        color: "#004BCC",
        textDecoration: "underline"
      }}
    />
  );
}

export { Anchor, P, Blockquote };
