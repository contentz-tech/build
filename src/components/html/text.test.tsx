/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { P, Blockquote, Anchor } from "./text";

describe("Text", () => {
  test("Paragraph", () => {
    const { asFragment, getByText } = render(
      <P>This is a test of a paragraph rendered by Contentz.</P>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/This is a test of a paragraph rendered by Contentz/))
      .toHaveTextContent("This is a test of a paragraph rendered by Contentz.");
  });

  test("Blockquote", () => {
    const { asFragment, getByText } = render(
      <Blockquote>
        This is a test of a paragraph rendered by Contentz.
      </Blockquote>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(getByText(/This is a test of a paragraph rendered by Contentz/))
      .toHaveTextContent("This is a test of a paragraph rendered by Contentz.");
  });

  test("Anchor Internal", () => {
    const { asFragment, getByText } = render(
      <Anchor href="/">Go to home</Anchor>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Go to home/)).toHaveTextContent("Go to home");
    expect(getByText(/Go to home/)).toHaveAttribute("target", "_self");
    expect(getByText(/Go to home/)).toHaveAttribute("href", "/");
  });

  test("Anchor External", () => {
    const { asFragment, getByText } = render(
      <Anchor href="https://contentz.tech">Contentz.tech</Anchor>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/Contentz.tech/)).toHaveTextContent("Contentz.tech");
    expect(getByText(/Contentz.tech/)).toHaveAttribute("target", "_blank");
    expect(getByText(/Contentz.tech/))
      .toHaveAttribute("rel", "nofollow noopener");
    expect(getByText(/Contentz.tech/))
      .toHaveAttribute("href", "https://contentz.tech");
  });
});
