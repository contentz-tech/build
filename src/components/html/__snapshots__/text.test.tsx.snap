// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Anchor External 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
    class="emotion-0"
    href="https://contentz.tech"
    rel="nofollow noopener"
    target="_blank"
  >
    Contentz.tech
  </a>
</DocumentFragment>
`;

exports[`Text Anchor Internal 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #004BCC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
    class="emotion-0"
    href="/"
    target="_self"
  >
    Go to home
  </a>
</DocumentFragment>
`;

exports[`Text Blockquote 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-left: 3px solid black;
  box-sizing: border-box;
  padding-left: calc(2em - 3px);
  margin-left: -2em;
  margin-right: 0;
  font-weight: normal;
  font-size: 1em;
}

.emotion-0 P {
  margin: 0;
}

<blockquote
    class="emotion-0"
  >
    This is a test of a paragraph rendered by Contentz.
  </blockquote>
</DocumentFragment>
`;

exports[`Text Paragraph 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 1em 0;
  font-size: 1em;
}

<p
    class="emotion-0"
  >
    This is a test of a paragraph rendered by Contentz.
  </p>
</DocumentFragment>
`;
